src/navigation/MainTabNavigator.tsx

```diff
@@ -1,37 +1,154 @@
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { colors, device, normalize } from '../config';
-import { TabNavigatorConfig } from '../types';
+import { LoadingSpinner } from '../components';
+import { colors, normalize } from '../config';
+import {
+  createDynamicTabConfig,
+  tabNavigatorConfig as defaultTabNavigatorConfig
+} from '../config/navigation/tabConfig';
+import { useStaticContent } from '../hooks';
+import { ScreenName, TabConfig, TabNavigatorConfig } from '../types';
 
 import { getStackNavigator } from './AppStackNavigator';
 
+const defaultTabRoutes: TabRoutes = {
+  tabConfig: {
+    activeBackgroundColor: colors.surface,
+    activeTintColor: colors.accent,
+    inactiveBackgroundColor: colors.surface,
+    inactiveTintColor: colors.primary,
+    tabs: ['Home', 'Service', 'Events', 'Index', 'About']
+  }
+};
+
+type Tab = {
+  accessibilityLabel: string;
+  iconName: string;
+  iconSize: number;
+  label: string;
+  screen: ScreenName;
+  params?: Record<string, any>;
+};
+
+type TabRoutes = {
+  tabConfig: {
+    activeTintColor: string;
+    inactiveTintColor: string;
+    activeBackgroundColor: string;
+    inactiveBackgroundColor: string;
+    tabs: (Tab | string)[];
+  };
+};
+
+const useTabRoutes = () => {
+  const { data, error, loading } = useStaticContent<TabRoutes>({
+    name: 'tabNavigation',
+    type: 'json'
+  });
+
+  const [tabRoutes, setTabRoutes] = useState<TabRoutes>(defaultTabRoutes);
+
+  useEffect(() => {
+    const tabs: (Tab | string)[] = data?.tabConfig.tabs || defaultTabRoutes.tabConfig.tabs;
+    const { activeBackgroundColor, activeTintColor, inactiveBackgroundColor, inactiveTintColor } =
+      data?.tabConfig || defaultTabRoutes.tabConfig;
+
+    setTabRoutes((prev) => {
+      const filteredTabs = defaultTabNavigatorConfig.tabConfigs.filter((tabConfig) => {
+        const screen = tabConfig.stackConfig.initialRouteName;
+
+        return tabs.some((tab) =>
+          typeof tab === 'string' ? tab === screen : tab.screen === screen
+        );
+      });
+
+      const dynamicTabs = tabs
+        .map((tab, index) => {
+          if (typeof tab === 'string') {
+            const screen = tab;
+            return filteredTabs.find(
+              (tabConfig) => tabConfig.stackConfig.initialRouteName === screen
+            );
+          } else {
+            return createDynamicTabConfig(
+              tab.accessibilityLabel,
+              tab.iconName,
+              tab.iconSize,
+              index,
+              tab.label,
+              tabs.length,
+              tab.screen,
+              tab.params
+            );
+          }
+        })
+        .filter(Boolean) as TabConfig[];
```

Comment:
why is this filter needed? can it happen, that there are undefined entries in the array? if yes, how?

Reply:
Actually, this is a filter that is added because objects can be 'null' or 'undefined' during development. But we don't need it at the moment.